Framework/Crystal/test/FindSXPeaksHelperTest.h

```diff
@@ -0,0 +1,394 @@
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidCrystal/FindSXPeaksHelper.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+#include "MantidKernel/ProgressBase.h"
+#include "MantidKernel/WarningSuppressions.h"
+#include <gmock/gmock.h>
+
+#include <string>
+
+using namespace Mantid::Crystal::FindSXPeaksHelper;
+using namespace testing;
+
+namespace {
+
+GCC_DIAG_OFF_SUGGEST_OVERRIDE
+
+class MockProgressBase : public Mantid::Kernel::ProgressBase {
+public:
+  MOCK_METHOD1(doReport, void(const std::string &));
+};
+
+GCC_DIAG_ON_SUGGEST_OVERRIDE
+}
+
+class FindSXPeaksHelperTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static FindSXPeaksHelperTest *createSuite() {
+    return new FindSXPeaksHelperTest();
+  }
+  static void destroySuite(FindSXPeaksHelperTest *suite) { delete suite; }
+
+  /* ------------------------------------------------------------------------------------------
+   * Single Cruystal peak representation
+   * ------------------------------------------------------------------------------------------
+   */
+
+  // Test out of bounds constuction arguments
+  void testSXPeakConstructorThrowsIfNegativeIntensity() {
+    auto workspace =
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(10, 10);
+    const auto &spectrumInfo = workspace->spectrumInfo();
+    double intensity = -1; // Negative intensity.
+    std::vector<int> spectra(1, 1);
+    TSM_ASSERT_THROWS("SXPeak: Should not construct with a negative intensity",
+                      SXPeak(0.001, 0.02, intensity, spectra, 0, spectrumInfo),
+                      std::invalid_argument);
+  }
+
+  // Test out of bounds construction arguments.
+  void testSXPeakConstructorThrowsIfSpectraSizeZero() {
+    auto workspace =
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(10, 10);
+    const auto &spectrumInfo = workspace->spectrumInfo();
+    double intensity = 1;
+    std::vector<int> spectra; // Zero size spectra list
+    TSM_ASSERT_THROWS(
+        "SXPeak: Should not construct with a zero size specral list",
+        SXPeak(0.001, 0.02, intensity, spectra, 0, spectrumInfo),
+        std::invalid_argument);
+  }
+
+  void testSXPeakGetters() {
+    auto workspace =
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(10, 10);
+    const auto &spectrumInfo = workspace->spectrumInfo();
+    double intensity = 1;
+    std::vector<int> spectra(1, 1);
+    SXPeak peak(0.001, 0.02, intensity, spectra, 1, spectrumInfo);
+
+    TSM_ASSERT_EQUALS("Intensity getter is not wired-up correctly", 1,
+                      peak.getIntensity());
+    TSM_ASSERT_EQUALS("Detector Id getter is not wired-up correctly", 2,
+                      peak.getDetectorId());
+    // QSpace is also a getter, but is tested more thouroughly below.
```

Comment:
This is code that I migrated from `FindSXPeaks` to the helper class. The tests existed beforehand. I will remove the comment.